packages/vue-searchbox/examples/with-transform-request-prop/src/App.vue

```diff
@@ -0,0 +1,204 @@
+                      <div class="book-header">{{ item.original_title }}</div>
+                      <div class="flex column justify-space-between">
+                        <div>
+                          <div>
+                            by
+                            <span class="authors-list">{{ item.authors }}</span>
+                          </div>
+                          <div class="ratings-list flex align-center">
+                            <span>
+                              <i
+                                v-for="(item, index) in Array(
+                                  item.average_rating_rounded
+                                ).fill('x')"
+                                class="fa fa-star stars"
+                                :key="index"
+                              />
+                            </span>
+                            <span class="avg-rating"
+                              >({{ item.average_rating_rounded }} avg)</span
+                            >
+                          </div>
+                        </div>
+                        <span class="pub-year"
+                          >Pub {{ item.original_publication_year }}</span
+                        >
+                      </div>
+                    </div>
+                  </div>
+                </div>
+                <div class="pagination-wrapper">
+                  <paginate
+                    v-if="results.numberOfResults"
+                    :value="Math.ceil(from / size) + 1"
+                    :page-count="Math.ceil(results.numberOfResults / size)"
+                    :click-handler="page => setFrom((page - 1) * size)"
+                    :prev-text="'Prev'"
+                    :next-text="'Next'"
+                    :container-class="'pagination'"
+                  ></paginate>
+                </div>
+              </div>
+            </div>
+          </search-component>
+        </div>
+      </div>
+    </search-base>
+  </div>
+</template>
+
+<script>
+/* eslint-disable no-console */
+import Paginate from 'vuejs-paginate';
+import {
+  SearchBase,
+  SearchComponent,
+  SearchBox
+} from '@appbaseio/vue-searchbox';
+import axios from 'axios';
+import './styles.css';
+
+export default {
+  name: 'app',
+  components: {
+    SearchBase,
+    SearchBox,
+    SearchComponent,
+    Paginate
+  },
+  data() {
+    return {
+      suggestedWords: []
+    };
+  },
+  methods: {
+    transformRequest(request) {
+      const suggestedWordsList = [];
+      let reqBody = JSON.parse(request.body);
+      let url =
+        'https://api.datamuse.com/words?sp=' +
+        reqBody.query[0].value +
+        '&max=2';
+      return axios
+        .get(url)
+        .then(({ data }) => {
+          if (data.length > 0) {
+            suggestedWordsList.push(data[0].word);
+          }
+          if (suggestedWordsList.length) {
+            reqBody.query[0].value = suggestedWordsList[0];
```

Comment:
Get query by component ID. It should be `search_component__internal` or check the request id when you type in input box.